test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetNewTemplatesTests.cs

```diff
@@ -0,0 +1,708 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using FluentAssertions;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace Dotnet.Integration.Test
+{
+    [Collection("Dotnet Integration Tests")]
+    public class DotnetNewTemplatesTests
+    {
+        private readonly MsbuildIntegrationTestFixture _fixture;
+
+        public DotnetNewTemplatesTests(MsbuildIntegrationTestFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Theory]
+        [InlineData("", "", "", false)]
+        [InlineData("ProjectA", "Myoutput", "VB", false)]
+        [InlineData("", "", "C#", true)]
+        [InlineData("ProjectA", "Myoutput", "F#", true)]
+        public void Dotnet_New_ConsoleApp_Success(string projectName, string output, string lang, bool norestore)
```

Comment:
Using `--no-restore` doesn't invoke NuGet in any way. Therefore, it's pointless to test in the NuGet repo.

Reply:
Ok. Addressed. Thank you for pointing this out.